### DISCUSSION TITLE
How to implement nuxt middleware correctly for auth?

### DISCUSSION THREAD

**Discussion by anshumankomawar** [vuejs/vuefire#1532] [Questions and Help]
`export default defineNuxtRouteMiddleware(async (to, from) => {
	const user = await getCurrentUser!();

	if (!user && to.path !== "/login") {
		return navigateTo({ path: "/login" });
	} else if (
		user &&
		(to.path === "/login" || to.path === "/register" || to.path === "/")
	) {
		return navigateTo({ path: "/home" });
	}
});
`

this is my global middleware. I believe it works but there is an issue where getCurrentUser is initially null and then its value changes to a user after some time. The user is already logged in and I have enabled session store and persistance. This behavior only happens on safari I didnt have an issue while testing on chrome. Safari routes to the login page for a second and then redirects to the home page. I'm new to vue so would appreciate any help.

**Comment by sugoidesune** [comment]
Theres getCurrentUser() that can be awaited.
https://vuefire.vuejs.org/nuxt/auth.html
```ts
export default defineNuxtRouteMiddleware(async (to, from) => {
  const user = await getCurrentUser()
    
    if (!user) {
      return navigateTo('/login')
    }
  })
  
  ```
